gix-url/src/lib.rs

```diff
@@ -78,6 +78,12 @@ pub enum ArgumentSafety<'a> {
 ///
 /// Additionally there is support for [deserialization](Url::from_bytes()) and [serialization](Url::to_bstring()).
 ///
+/// # Mutability Warning
+///
+/// Due to the mutability of this type, it's possible that the URL serializes to something invalid
+/// when fields are modified directly. The URL should be accepted as a parameter directly rather
```

Comment:
Make clear that the URL to accept in parameters should always be parsed to this type, but never be an instance of this type.